controllers/managedocs_controller_test.go

```diff
@@ -2,639 +2,759 @@ package controllers
+			Namespace: testPrimaryNamespace,
+		},
+		Status: &promv1.PrometheusStatus{},
+	}
+	amTemplate := promv1.Alertmanager{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      alertmanagerName,
+			Namespace: testPrimaryNamespace,
+		},
+		Status: &promv1.AlertmanagerStatus{},
+	}
+	podMonitorTemplate := promv1.PodMonitor{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-pod-monitor",
+			Namespace: testPrimaryNamespace,
+		},
+		Spec: promv1.PodMonitorSpec{
+			PodMetricsEndpoints: []promv1.PodMetricsEndpoint{},
+		},
+	}
+	serviceMonitorTemplate := promv1.ServiceMonitor{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-service-monitor",
+			Namespace: testPrimaryNamespace,
+		},
+		Spec: promv1.ServiceMonitorSpec{
+			Endpoints: []promv1.Endpoint{},
+		},
+	}
+	promRuleTemplate := promv1.PrometheusRule{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-prometheus-rule",
+			Namespace: testPrimaryNamespace,
+		},
+	}
+	addonParamsSecretTemplate := corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      testAddonParamsSecretName,
+			Namespace: testPrimaryNamespace,
+		},
+		Data: map[string][]byte{},
+	}
+	addonConfigMapTemplate := corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      testAddonConfigMapName,
+			Namespace: testPrimaryNamespace,
+		},
+	}
+	pvc1StorageClassName := storageClassRbdName
+	pvc1Template := corev1.PersistentVolumeClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-pvc-1",
+			Namespace: testPrimaryNamespace,
+		},
+		Spec: corev1.PersistentVolumeClaimSpec{
+			StorageClassName: &pvc1StorageClassName,
+			AccessModes: []corev1.PersistentVolumeAccessMode{
+				corev1.ReadWriteOnce,
+			},
+			Resources: corev1.ResourceRequirements{
+				Requests: corev1.ResourceList{
+					corev1.ResourceStorage: resource.MustParse("1Gi"),
+				},
+			},
+		},
+	}
+	pvc2StorageClassName := storageClassCephFSName
+	pvc2Template := corev1.PersistentVolumeClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-pvc-2",
+			Namespace: testPrimaryNamespace,
+		},
+		Spec: corev1.PersistentVolumeClaimSpec{
+			StorageClassName: &pvc2StorageClassName,
+			AccessModes: []corev1.PersistentVolumeAccessMode{
+				corev1.ReadWriteOnce,
+			},
+			Resources: corev1.ResourceRequirements{
+				Requests: corev1.ResourceList{
+					corev1.ResourceStorage: resource.MustParse("1Gi"),
+				},
+			},
+		},
+	}
+	subscriptionTemplate := opv1a1.Subscription{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      testSubscriptionName,
+			Namespace: testPrimaryNamespace,
+		},
+	}
+	csvTemplate := opv1a1.ClusterServiceVersion{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      testDeployerCSVName,
+			Namespace: testPrimaryNamespace,
+		},
+	}
+
+	setupUninstallConditions := func(
+		shouldAddonConfigMapExist bool,
+		addonConfigMapDeleteLabel string,
+		shouldStorageClusterBeReady bool,
+		shouldPrometheusBeReady bool,
+		shouldAlertmanagerBeReady bool,
+		shouldPVC1Exist bool,
+		shouldPVC2Exist bool,
+	) {
```

Comment:
[suggestion] I like this function. I would like to suggest encapsulating the parameters into a struct so that they are explicitly listed when the function is called. I had to keep hopping back to the definition to check which boolean value went to which parameter. Plus if the list where to ever grow, having it in a struct would mean you wouldn't have to modify every call to this function.

Reply:
Good suggestion, but this is a change that can be introduced in a future PR

Reply:
agreed.